contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditSourceModal.vue

```diff
@@ -0,0 +1,313 @@
+
+  function generateGetterSetter(key) {
+    return {
+      get() {
+        if (this[key] === nonUniqueValue) {
+          return this.$tr('mixed');
+        }
+        return this[key];
+      },
+      set(value) {
+        this[key] = value;
+      },
+    };
+  }
+
+  function mapFormGettersSetters(keys) {
+    return keys.reduce((acc, key) => {
+      acc[`${key}_value`] = generateGetterSetter(key);
+      return acc;
+    }, {});
+  }
+
+  const sourceKeys = [
+    'author',
+    'provider',
+    'aggregator',
+    'license',
+    'license_description',
+    'copyright_holder',
+  ];
+
+  export default {
+    name: 'EditSourceModal',
+    components: {
+      HelpTooltip,
+      LicenseDropdown,
+    },
+    props: {
+      nodeIds: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      /* eslint-disable  kolibri/vue-no-unused-properties */
+      return {
+        author: '',
+        provider: '',
+        aggregator: '',
+        license: '',
+        license_description: '',
+        copyright_holder: '',
+        licenseError: '',
+        licenseDescriptionError: '',
+        copyrightHolderError: '',
+      };
+    },
+    computed: {
+      ...mapGetters(['isAboutLicensesModalOpen']),
+      ...mapGetters('contentNode', ['getContentNodes']),
+      ...mapFormGettersSetters(sourceKeys),
+      nodes() {
+        return this.getContentNodes(this.nodeIds);
+      },
+      licenseItem: {
+        get() {
+          const { license, license_description } = this;
+          return {
+            license,
+            license_description,
+          };
+        },
+        set(value) {
+          this.license = value.license;
+          this.license_description = value.license_description;
+        },
+      },
+      isEditable() {
+        return this.nodes.some(node => !isDisableSourceEdits(node));
+      },
+      hasSomeEditDisabled() {
+        return this.nodes.some(node => isDisableSourceEdits(node));
+      },
+      hasError() {
+        return this.licenseError || this.licenseDescriptionError || this.copyrightHolderError;
+      },
+      helpText() {
+        if (!this.isEditable) {
+          return this.$tr('cannotEditPublic');
+        }
+        if (this.hasSomeEditDisabled) {
+          return this.$tr('editOnlyLocal');
+        }
+        return '';
+      },
+    },
+    created() {
+      const values = {};
+      this.nodes.forEach(node => {
+        sourceKeys.forEach(prop => {
+          if (node[prop]) {
+            if (!values[prop]) {
+              values[prop] = node[prop];
+            } else if (values[prop] !== node[prop]) {
+              values[prop] = nonUniqueValue;
+            }
+          }
+        });
+      });
+      sourceKeys.forEach(prop => {
+        this[prop] = values[prop] || '';
+      });
+    },
+    methods: {
+      ...mapActions('contentNode', ['updateContentNode']),
+      close() {
+        this.$emit('close');
+      },
+      validate() {},
```

Comment:
Is the validate meant to do anything?

Reply:
Hi! @akolson. Yes, It was meant to validate the attribution :sweat: I have updated the code! Could you please review it again? :)